### PR TITLE
fix: two questions, involving four rules

### PR DESCRIPTION
**Pull request by winfredLIN**
#1974 
### two question:
1. the rule DMLCheckFuzzySearch unsupport upper function
2. when there is no where in the outer layer of the SQL statement, the where statement of the subquery will not be checked
### involve four rules:
  1. DMLCheckWhereIsInvalid
  2. DMLCheckWhereExistNot
  3. DMLCheckWhereExistScalarSubquery
  4. DMLCheckFuzzySearch
### Solution
1. Rule DMLCheckFuzzySearch supports the upper function
2. When there is no WHERE in the outer layer of the SQL statement but there is a WHERE statement in its subquery, the WHERE statement in the subquery will also be checked

### CODE REVIEW

File: sqle/driver/mysql/rule/rule.go [hunk 1]
```diff
@@ -3121,53 +3121,57 @@ func isSelectCount(selectStmt *ast.SelectStmt) bool {
...
-		addResult(res, rule, DMLCheckFuzzySearch)
-		isAddResult = true
+	for _, where := range whereList {
+		if util.CheckWhereFuzzySearch(where) {
+			addResult(res, rule, DMLCheckFuzzySearch)
+			break
+		}
 	}
-	return isAddResult
+
```
Comment (hasa1K) [inline sqle/driver/mysql/rule/rule.go:3174]: 这4个for循环为什么不能合起来?

Comment (winfredLIN) [reply sqle/driver/mysql/rule/rule.go:3174]: 拆开来的原因是，当一条规则触发了，就不需要再检查了，直接退出循环
如果要合起来也可以，就要增加四个变量：
- 当一个规则触发了，设置为true
- 并且在每一个规则判断之前判断该规则是否已经触发，
- 并且在循环中增加判断是否四个规则都触发了，若触发则退出循环

File: sqle/driver/mysql/util/parser_helper.go [hunk 2]
```diff
@@ -410,18 +410,13 @@ func CheckWhereFuzzySearch(where ast.ExprNode) bool {
...
-						switch pattern := pattern.Args[0].(type) {
-						case *driver.ValueExpr:
-							datum := pattern.Datum.GetString()
-							if strings.HasPrefix(datum, "%") || strings.HasPrefix(datum, "_") {
-								isExist = true
-								return true
-							}
-						}
-					}
+				result := NewFuncCallStringResultGenerator(pattern).GenerateResult()
```
Comment (hasa1K) [inline sqle/driver/mysql/util/parser_helper.go:413]: 对upper，contact函数缺少单测

Comment (winfredLIN) [reply sqle/driver/mysql/util/parser_helper.go:413]: Get

Comment (winfredLIN) [reply sqle/driver/mysql/util/parser_helper.go:413]: done

**Review by hasa1K — APPROVED** [review]

**Pull request merged**